MIPS/mip-0005-mip-upgrade.md

```diff
@@ -0,0 +1,315 @@
+- Champion the MIP through each stage of the process.
+- Designate, coordinate and oversee a Dev Team for the MIP implementation.
+
+**Dev Team:** core development team, contributors and any other active participants that are making updates, parameter changes, and upgrades to the protocol itself. 
+
+Responsibilities:
+- Review MIP’s implementation plan.
+- Execute the implementation plan of approved MIPs.
+
+Functions:
+- Assess the feasibility of the MIP’s implementation plan and provide feedback to the MIP Author.
+- Ensure correct execution of the implementation plan in a timely manner.
+- Communicate and explain any delays and other important decisions about the implementation.
+
+#### Introducing New Roles and Responsibilities:
+
+**MIP Facilitator:** Elected community member with a deep understanding of Mina Protocol’s governance to oversee that the MIP process is followed correctly.
+
+Responsibilities:
+- Ensure a smooth progression of MIPs through the process.
+- Facilitate cooperation and collaboration between all participants and ensure they follow the rules.
+- Assist and support MIP Authors as necessary.
+
+Functions:
+- Accept or reject a MIP submission based on the established approval criteria.
+- Accept or reject a status change request ensuring that each step of the process is completed correctly.
+- Schedule and facilitate meetings between MIP Author and MIP Review Group, and other stakeholders.
+- Produce and share meeting minutes under the Chatham House Rule.
+- Identify key stakeholders affected by the MIP.
+- Coordinate communications with the Mina Foundation.
+
+**MIP Reviewer:** Elected community members who have expertise to review a MIP in a specific category.
+
+Responsibilities:
+- Review the assigned MIP.
+- Provide feedback to the Author to improve the MIP.
+
+Functions:
+- Identify risks, benefits and tradeoffs of the proposed changes.
+- Assess the risk of a MIP to the long-term health of the protocol.
+- Publish a Risk Assessment with a rationale.
+- Join the meetings scheduled by the MIP Facilitator.
+
+#### New Role Frameworks
+
+To ensure accountability and transparency for MIP Facilitators and Reviewers, we propose developing frameworks that clearly explain the processes for role selection, assignment, removal, resignation, and compensation. 
+
+These frameworks will share common principles, including:
+
+- Any community member can propose themselves as a candidate for MIP Facilitator or MIP Reviewer by submitting a new MIP.
+- Any community member can propose the removal of an existing MIP Facilitator or MIP Reviewer by submitting a new MIP.
+- Any community member can vote to approve or reject the selection or removal of MIP Facilitators and MIP Reviewers.
+- As the community builds its capacity to effectively take on these responsibilities, key ecosystem players will continue to fulfill these roles as needed, supporting the bootstrapping and testing of the new processes.
+
```

Comment:
Should we add the possibility for an MIP facilitator/reviewer to step down?

Reply:
Yes, I’ll add the possibility for an MIP Facilitator/Reviewer to step down in both this document and the attached Facilitator/Reviewer frameworks.

Additionally, I’ll clarify more explicitly that the roles of MIP Author, Reviewer, and Facilitator must be held by different individuals for any specific MIP. This separation of responsibilities ensures impartiality and avoids potential conflicts of interest.